src/analyzers/type-refactoring/discriminated-union-transformer.ts

```
@@ -0,0 +1,641 @@
+
+    try {
+      // Generate helper functions
+      const helpers = candidate.transformationPlan.generatedCode;
+      
+      // Add type guards
+      for (const typeGuard of helpers.typeGuards) {
+        this.appendCode(sourceFile, typeGuard);
+      }
+
+      // Add constructors
+      for (const constructor of helpers.constructors) {
```

Biome (2.1.2)

[error] Do not shadow the global "constructor" property.

Consider renaming this variable. It's easy to confuse the origin of variables when they're named after a known global.

(lint/suspicious/noShadowRestrictedNames)